libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt

```diff
@@ -96,21 +112,46 @@ class NavigationRoute internal constructor(
         ): List<NavigationRoute> {
             return routeParser.parseDirectionsResponse(
                 directionsResponseJson, routeOptionsUrlString
-            ).fold(
-                { error ->
-                    throw RuntimeException("Failed to parse a route. Reason: $error")
-                },
-                { value ->
-                    value.mapIndexed { index, routeInterface ->
-                        NavigationRoute(
-                            directionsResponse,
-                            index,
-                            routeOptions,
-                            routeInterface
-                        )
-                    }
-                }
-            ).cache()
+            ).run {
+                create(this, directionsResponse, routeOptions)
+            }
+        }
+
+        private suspend fun create(
+            directionsResponseJson: String,
+            routeOptions: RouteOptions,
+            routeOptionsUrlString: String,
+            routeParser: SDKRouteParser = NativeRouteParserWrapper
+        ): List<NavigationRoute> {
+            val deferredDirectionsResponse = CoroutineScope(Dispatchers.Default).async {
+                DirectionsResponse.fromJson(directionsResponseJson)
+            }
+            return routeParser.parseDirectionsResponse(
+                directionsResponseJson,
+                routeOptionsUrlString
+            ).run {
+                create(this, deferredDirectionsResponse.await(), routeOptions)
+            }
+        }
```

Comment:
I couldn't remember that syntax and cancelling didn't seem important to me since the original implementation didn't have any cancellation.  I"ll make an adjustment.